spring-amqp/src/main/java/org/springframework/amqp/remoting/service/AmqpInvokerServiceExporter.java

```diff
@@ -79,11 +79,12 @@ public void onMessage(Message message) {
 			RemoteInvocation invocation = (RemoteInvocation) invocationRaw;
 			remoteInvocationResult = invokeAndCreateResult(invocation, getService());
 		}
-		send(remoteInvocationResult, replyToAddress);
+		send(remoteInvocationResult, replyToAddress, message);
 	}
 
-	private void send(Object object, Address replyToAddress) {
+	private void send(Object object, Address replyToAddress, Message requestMessage) {
 		Message message = this.messageConverter.toMessage(object, new MessageProperties());
+		message.getMessageProperties().setCorrelationId(requestMessage.getMessageProperties().getCorrelationId());
```

Comment:
Well, really, the client has to be Spring AMQP - it's using Spring Remoting over AMQP.

> do we have...

Yes; the `RemotingTests` failed with this (using a DRTMLC requires the correlationId).

Yes, it's a bug and should be back-ported.